Adding different sizes for wooden wheel variants, and wooden casters set

Leha44581 opened the pull request:
#### Summary
Content "Added a set of wooden caster wheels, as a crude replacement to shopping cart caster wheels Also adds 8 inch, 12 inch, 16 inch wooden wheels"
#### Purpose of change

I think the player should be able to make more than just one size of wooden wheels

#### Describe the solution

Edited some jsons to add the new recipes/items/vehicleparts

#### Testing

The recipes seem to work fine, i have successfully repaired one shopping cart

#### Additional context
Wooden caster wheels are very much a real thing, and are often sold as "antique"
![Antique+Cast+Iron+Industrial+Furniture+Stem+Casters+1b](https://github.com/user-attachments/assets/3995c458-5e7b-4270-b771-9655dd8bb8e3)

Leha44581 committed:
Adding different sizes for wooden wheel variants

Added a set of wooden caster wheels, as a potential crude replacement to shopping cart caster wheels
Also adds 8 inch, 12 inch, 16 inch wooden wheels

Leha44581 committed:
Update wheel.json

QuillInkwell commented on data/json/items/vehicle/wheel.json:
```diff
@@ -760,6 +760,103 @@
+    "description": "A handmade 8 inch wooden cart wheel with metal bands for durability.",
+    "relative": { "weight": "100 g" }
+  },
+    {
```
```suggestion
  {
```

moxian commented:
The image you attach in the OP does not depict wooden casters - they are pretty clearly metallic, although the lightning makes it a bit ambiguous. Here's another shot of the same antique piece (from https://www.ironanarchy.com/furniture-casters/p/antique-furniture-stem-casters-double-wheel-3sj5j)
<details>

 ![Antique+Cast+Iron+Industrial+Furniture+Stem+Casters+1a](https://github.com/user-attachments/assets/c9f3385c-e97e-413e-b4fe-1b750eb19a65)

</details>

From the same website, another set of casters, these ones are wooden but are definitely still in a metal frame: https://www.ironanarchy.com/furniture-casters/p/antique-swivel-furniture-casters-fkz57 

<details>

![Small+Antique+Cast+Iron+Stem+Furniture+Caster+Wheels,+Schenck+1a](https://github.com/user-attachments/assets/c2e4c55d-9448-4612-a6ef-60da0407f256)

</details>

Even assuming the housing was wooden, it looks a bit bigger than a 2x4 plank so can't be exactly carved out of it? I think? Also why does that recipe need a HAMMER quality?

----

Completely completely separately

> I think the player should be able to make more than just one size of wooden wheels

Do you any of these different-sized wheels (aside from casters which are a different topic) be ever used?
As a player myself I would either 1) be confused about why do we have so many different sizes 2) wonder which size i need and 3) either choose the size that's "most optimal" (surely there's a most optimal one?) or 4) choose one completely at random. But in either case this choice feels pretty unsatisfying.
I could perhaps be fine with the different sized wooden if those were *found* items, where I don't have much of a choice and I'm just using whatever I have (and i don't care!), until i have something better (like what happens with half-wrecked cars and their parts). But this approach does not quite work when the player is forced to make the choice from the start.

Leha44581 committed:
Formatting changes, removed hammed requirement for caster wheels

Leha44581 committed:
Update data/json/recipes/recipe_vehicle.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Leha44581 committed:
Update data/json/vehicleparts/wheel.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Leha44581 committed:
Update data/json/vehicleparts/wheel.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Leha44581 committed:
Update data/json/vehicleparts/wheel.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Leha44581 committed:
Update data/json/vehicleparts/wheel.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

GuardianDll committed:
lint

GuardianDll commented:
(~[slow] ~[.],starting_items)=> (continued from above) ERROR : src/assign.cpp:19 [void report_strict_violation(const JsonObject &, const std::string &, const std::string_view)] (json-error)
Error: (~[slow] ~[.],starting_items)=> cannot assign explicit value the same as default or inherited value

    "difficulty": 4,
    "time": "4 h",
    "reversible": false,
                ▲▲▲
    "autolearn": true,
    "proficiencies": [ { "proficiency": "prof_carpentry_basic" } ],


(~[slow] ~[.],starting_items)=> 08:04:10.929 INFO : Fail to initialize global game state
Error: (~[slow] ~[.],starting_items)=> Must use denomination for pricing.  Valid units are:
cents
cent
dollars
dollar
USD
kUSD

    "volume": "1500 ml",
    "longest_side": "110 mm",
    "price": 20000,
           ▲▲▲
    "price_postapoc": 50,
    "to_hit": -1,

Leha44581 committed:
Removed 8 inch wheels, changed other added wheels.

- 8 Inch wooden wheels removed

- 12 Inch wooden wheels renamed into wheelbarrow wheels to not overwhelm the player, these are stable so an actual single wheeled wheelbarrow can be made

- 16 Inch wooden wheels renamed into small wooden cart wheels

- Added a hole drill requirement for all added wheels

Leha44581 committed:
Changed pricing, syntax fixes

GuardianDll committed:
resolve errors

GuardianDll commented:
```
[
  "wheel_wood_16i",
  "wheel_wood_12i",
  "wheel_wood_caster"
]
The item(s) above use legacy to_hit, please change them to the newer object
method (see /docs/design-balance-lore/GAME_BALANCE.md#to-hit-value) or remove
the to_hit field if the item(s) aren't intended to be used as weapons."
```
```
(~[slow] ~[.],starting_items)=>   CHECK( item_density( target ) <= max_density )
(~[slow] ~[.],starting_items)=> with expansion:
Error: (~[slow] ~[.],starting_items)=>   1.0f <= 0.85f
(~[slow] ~[.],starting_items)=> with messages:
(~[slow] ~[.],starting_items)=>   target.typeId() := string_id( "wheel_wood_caster" )
(~[slow] ~[.],starting_items)=>   target.weight() := 1500000mg
(~[slow] ~[.],starting_items)=>   target.volume() := 1500ml
```

Maddremor commented:
Have you ever wondered why old fashioned wagon wheels are so large compared to the smaller wheels we have on cars? It's because the smaller the wheel, the faster it needs to rotate for a given speed, and this is a problem when you don't have any fancy modern bearings to help you. A smaller wooden wheel might be reasonable, but I wouldn't put it on something like a caster since that would add mechanical complexity. A fixed wheel like that of a wheelbarrow can work for most circumstances. 

If you really want to add something small-ish to make a primitive hand cart easier to drag around, then a sled runner of some kind would be the way to go, but for that we need to add flags or properties that keeps it from being used for steering or propulsion.
